zebra-network/src/peer/client.rs

```diff
@@ -199,10 +189,7 @@ impl Service<Request> for Client {
 
     fn poll_ready(&mut self, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
         if ready!(self.server_tx.poll_ready(cx)).is_err() {
-            Poll::Ready(Err(self
-                .error_slot
-                .try_get_error()
-                .expect("failed servers must set their error slot")))
```

Comment:
This change replaces `error_slot` with `PeerError::ConnectionClosed`.

What happened to the error value that used to be here?
Are we hiding more specific errors as part of this change?

Reply:
possibly, but I think the errors we used to grab here were misleading and shouldn't be propagated back to this source. The error slot used to store the error that brought down a `Connection`, but that error is actually associated with one of the client requests created by `call`. We would propagate the error back to that caller over the channel and also copy it to be shared with all future callers. My instinct is that this isn't particularly useful and that we will correctly report all these errors still, they just wont end up being reported many extra times. Imo its fine for subsequent attempts to use a `Client` with a dead `Connection` to just say "sorry this is already closed".